### PR TITLE
store/tikv: remove Peer

### PR DESCRIPTION
**Pull request by disksing**
@siddontang @shenli @morefreeze

### CODE REVIEW

File: store/tikv/region_cache.go [hunk 1]
```diff
@@ -57,72 +57,69 @@ func (c *RegionCache) DropRegion(regionID uint64) {
...
 		store *metapb.Store
 		err   error
 	)
 
-	curPeerIdx := -1
-	for idx, p := range old.meta.Peers {
-		if p.GetId() == leaderID {
-			peer = p
+	curStoreIdx := -1
+	for idx, id := range old.meta.StoreIds {
```
Comment (morefreeze) [inline store/tikv/region_cache.go]: `storeID` seems better. `id`, `idx` are both confused.

**Comment by coocood** [comment]
LGTM

File: store/tikv/region_cache_test.go [hunk 2]
```diff
@@ -191,28 +191,28 @@ func (s *testRegionCacheSuite) TestReconnect(c *C) {
 func (s *testRegionCacheSuite) TestNextPeer(c *C) {
```
Comment (coocood) [inline store/tikv/region_cache_test.go]: TestNextStore

**Comment by shenli** [comment]
LGTM

**Comment by morefreeze** [comment]
LGTM

**Comment by disksing** [comment]
deps updated, PTAL @coocood @siddontang

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**